### DISCUSSION TITLE
Guide to (somewhat) running on windows.

### DISCUSSION THREAD

**Discussion by LAP87** [rivenmedia/riven#645] [General]
**SYMLINKING PROBLEMATIC/NOT CURRENTLY WORKING**

Install Git, Node.js and Python 3 first if you haven't already.
Git: https://git-scm.com/downloads
Node.js https://nodejs.org/en/download/prebuilt-installer/current
Python 3.12+ https://www.python.org/downloads/ and make sure to add it to PATH during install.

_Instructions for PostgreSQL by Bartle (w/ modifications from me)_
**To set up a Windows environment to use the connection string `postgresql+psycopg2://postgres:postgres@localhost/riven` with SQLAlchemy, follow these steps:**

### 1. **Install PostgreSQL**
   - Download PostgreSQL from the [official PostgreSQL website](https://www.postgresql.org/download/windows/).
   - Run the installer and follow the prompts. During the installation:
     - Make sure to install `pgAdmin` (optional but useful).
     - Remember the password you set for the `postgres` user (you can use `postgres` to match your connection string).
     - Install the Stack Builder if you plan to use additional extensions.

   - Once the installation is complete, PostgreSQL should be running as a Windows service.

### 2. **Set Up the Database and User**
   - Open the PostgreSQL command line (you can find it in the Start menu as "SQL Shell (psql)").
   - Connect to the default `postgres` database by pressing enter on each option until you get to terminal.

- Now, create the `riven` database:

     ```sql
     CREATE DATABASE riven;
     ```

   - You don't need to create the `postgres` user since it was created during installation, but you need to reset the password:

     ```sql
     ALTER USER postgres WITH PASSWORD 'postgres';
     ```

   - Exit the psql prompt:

     ```sql
     \q
     ```

### 3. **Install Dependencies**
   - Open Command Prompt and install the `psycopg2` and `SQLAlchemy` libraries:

     ```bash
     pip install psycopg2-binary sqlalchemy
     ````

**If you encounter UTF-8 errors, follow this guide on how to fix it.**
https://stackoverflow.com/a/57134096

- Reboot.

- open CMD/Powershell in C: drive twice.

Shell -1
```
git clone https://github.com/rivenmedia/riven.git && cd riven
pip install poetry
poetry install --without dev
poetry run python src/main.py
```

_This will launch the backend, you can minimize this window now and continue with the frontend._

Shell -2
```
git clone https://github.com/rivenmedia/riven-frontend.git && cd riven-frontend
npm install
$env:ORIGIN = 'http://localhost:3000'
$env:BACKEND_URL = 'http://localhost:8080'
$env:DIALECT = 'postgres'
$env:DATABASE_URL = 'postgresql://postgres:postgres@localhost/riven'
node build
```

_This launches the frontend and spits out a localhost:XXXX adress where you can access riven from any browser.
Minimize this window also. (If you accidentally leftclick anywhere on the prompts it will pause the program, press ESC once to resume)
Someone might be able to figure out how to easily make these run hidden from boot, but that is not me._

**Comment by dreulavelle** [comment]
the issue isnt getting the app to run on windows, its plex that doesnt like the symlinks. Maybe JF and Emby would work with the windows setup

**Comment by LAP87** [comment]
> the issue isnt getting the app to run on windows, its plex that doesnt like the symlinks. Maybe JF and Emby would work with the windows setup

Yeah, it stopped running after completing the setup and crashes when i try to restart it.. dang!
[riven-20240814-0133.log](https://github.com/user-attachments/files/16606443/riven-20240814-0133.log)

**Comment by dreulavelle** [comment]
those crashes look likes Riven cant find the database. Make sure postgres is installed

**Comment by LAP87** [comment]
> > those crashes look likes Riven cant find the database. Make sure postgres is installed
> 
> Installed the full suite PostgresQL for win 64 and rebooted, now there's a different UTF-8 related error, maybe you understand it better? 
> [riven-20240814-1729.log](https://github.com/user-attachments/files/16615507/riven-20240814-1729.log)
> 
I fixed it with some help from the discord, the guide is updated with all steps i've taken.
The issue next is figuring out the whole symlinking situation on windows.

**Comment by AyushSehrawat** [comment]
Switched it to a discussion

**Comment by AyushSehrawat** [comment]
hardlinks work on windows but @dreulavelle had some issues with it i guess

**Reply by dreulavelle** [reply]
I can give it another try ig.

**Comment by truncsphere** [comment]
Any progress on a windows version? I have Plex running on a Windows VM and a docker host. Setting it this up on the same server as Plex would greatly simplify the setup/configurations.

**Comment by profucius** [comment]


**Reply by AyushSehrawat** [reply]
> I too would like to see progress in the windows version. Plex being already installed in windows with Riven installed along side it in OS would save me so much time that I don't have from switching it to a docker.

It's unlikely there will be any progress on windows because of symlink not working correctly with plex in it
